api/src/utils/hooks.ts

```diff
@@ -52,33 +48,19 @@ export async function verifyJWT(context: HookContext) {
     return Promise.reject(new Forbidden("The authorization header is missing"));
   }
   const token = authHeader.split(" ")[1];
-
-  // fetch public key from JWKS url and verify token
-  const jwksOptions = {
-    jwksUri: context.app.get("authentication").jwksUri,
-  } as ClientOptions;
-  const client = jwks(jwksOptions) as JwksClient;
-  const keys = (await client.getSigningKeysAsync()) as SigningKey[];
-
-  let decoded;
-  try {
-    decoded = verify(token, keys[0].getPublicKey(), {
-      algorithms: context.app.get("authentication").algorithms,
-    });
-  } catch (error) {
-    throw new NotAuthenticated(`Authentication failed: ${error.message}`);
-  }
+  const keys = await getAuthSigningKeys(context);
+  decodeIdToken(token, keys, context);
   return context;
 }
 
 export function verifyJWTRoles(roles: string[]) {
   return async (context: HookContext) => {
-    const authHeader = context.params.headers?.authorization as string;
-    if (!authHeader) {
+    const token = extractIdToken(
+      context.params.headers?.authorization as string
+    );
+    if (!token) {
       throw new Forbidden("The authorization header is missing");
     }
-    const token = authHeader.split(" ")[1];
-
     const decoded = decode(token) as {
```

Comment:
What happens to the block of code following this if `decoded === undefined`

Reply:
This should raise an exception that would break the chain and fail the request.